docs/Hub.md

```diff
@@ -0,0 +1,210 @@
+ object for use with the ONNX runtime.
+
+```python
+model = hub.load("resnet50")
+```
+
+
+#### Downloading from custom repositories:
+
+Any repository with the proper structure can be a ONNX model hub. To download from other hubs,
+ or to specify a particular branch or commit on the main model hub one can provide the `repo` parameter:
+
+```python
+model = hub.load("resnet50", repo='onnx/models:771185265efbdc049fb223bd68ab1aeb1aecde76')
+```
+
+#### Listing and inspecting Models:
+
+The model hub provides APIs for querying the model zoo to learn more about available models.
+ This does not download the models, but rather just returns information about models matching the given arguments
+ 
+ ```python
+# List all models in the onnx/models:master repo
+all_models = hub.list_models()
+
+# List all versions/opsets of a specific model
+mnist_models = hub.list_models(model="mnist")
+
+# List all models matching a given "tag"
+vision_models = hub.list_models(tags=["vision"])
+```
+
+One can also inspect the metadata of a model prior to download with the `get_model_info` function:
+
+```python
+print(hub.get_model_info(model="mnist", opset=8))
+```
+This will print something like:
+```
+ModelInfo(
+    model=MNIST,
+    opset=8,
+    path=vision/classification/mnist/model/mnist-8.onnx,
+    metadata={
+     'model_sha': '2f06e72de813a8635c9bc0397ac447a601bdbfa7df4bebc278723b958831c9bf',
+     'model_bytes': 26454,
+     'tags': ['vision', 'classification', 'mnist'],
+     'io_ports': {
+        'inputs': [{'name': 'Input3', 'shape': [1, 1, 28, 28], 'type': 'tensor(float)'}],
+        'outputs': [{'name': 'Plus214_Output_0', 'shape': [1, 10], 'type': 'tensor(float)'}]},
+     'model_with_data_path': 'vision/classification/mnist/model/mnist-8.tar.gz',
+     'model_with_data_sha': '1dd098b0fe8bc750585eefc02013c37be1a1cae2bdba0191ccdb8e8518b3a882',
+     'model_with_data_bytes': 25962}
+)
+```
+
+
+## Local Caching
+
+The ONNX Model hub locally caches downloaded models in a configurable location 
+so that subsequent calls to `hub.load` do not require network connection.
+
+#### Default cache location
+The hub client looks for the following default cache locations in this order:
+
+1) `$ONNX_HOME/hub` if the `ONNX_HOME` environment variable is defined
+2) `$XDG_CACHE_HOME/hub` if the `XDG_CACHE_HOME` environment variable is defined
+3) `~/.cache/onnx/hub` where `~` is the user home directory
+
+#### Setting the cache location
+To manually set the cache location use:
+
+```python
+hub.set_dir("my/cache/directory")
+```
+
+Additionally one can inspect the cache location with:
+
+```python
+print(hub.get_dir())
+```
+
+#### Additional cache details
+
+To clear the model cache one can simply delete the cache directory using a python utility like `shutil` or `os`. 
+Furthermore one can choose to override the cached model using the `force_reload` option:
+
+```python
+model = hub.load("resnet50", force_reload=True)
```

Comment:
add a best practice note to set force_reload=False to avoid unnecessary bandwidth cost.